cartographer/mapping/internal/2d/scan_matching/ceres_scan_matcher_2d.cc

```diff
@@ -102,6 +100,18 @@ void CeresScanMatcher2D::Match(const Eigen::Vector2d& target_translation,
 
   ceres::Solve(ceres_solver_options_, &problem, summary);
 
+  if(seedless_precision)
+  {
+      auto parameter_blocks   = std::array<const double*, 1>{{ceres_pose_estimate}};
+      auto residuals          = std::vector<double>(point_cloud.size(), 0.0);
+      auto row_major_jacobian = Eigen::Matrix<double, Eigen::Dynamic, 3, Eigen::RowMajor>(point_cloud.size(), 4);
+      auto jacobianPtr        = row_major_jacobian.data(); // for use to create fake array of jacobians
+
+      costFunction->Evaluate(parameter_blocks.data(), residuals.data(), &jacobianPtr);
```

Comment:
Not of any concern here, but there is also `problem.Evaluate` method that can populate residuals and jacobians using the final solution. FYI

Reply:
I didn't use that one because the problem still has the seed in it. As far as I've read removing a cost function can mess up the jacobians.